src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -21,8 +21,12 @@ class Serializer implements DecoderInterface, EncoderInterface
     private $serializer;
     private $format;
 
-    public function __construct(SerializerInterface $serializer, string $format = 'json')
+    public function __construct(SerializerInterface $serializer = null, string $format = 'json')
     {
+        if (null === $serializer) {
+            throw new \InvalidArgumentException('A Serializer is required to use this Messenger transport. Try running "composer require symfony/serializer".');
```

Comment:
IMO it's not the responsibility of this class to do that. It should be part of the DI pass, where it was handled before.

Reply:
I agree. The benefit is that it'd remains a compile-time check. Here it's only at runtime. Also, it's not necessarily the component that is not installed. It may just not be enabled.